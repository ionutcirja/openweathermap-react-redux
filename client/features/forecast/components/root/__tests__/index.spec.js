import React from 'react';
import { shallow } from 'enzyme';
import { Component } from '..';

describe('Forecast component', () => {
  let propsToRender;
  beforeEach(() => {
    propsToRender = {
      list: {},
      theme: {
        colours: {
          greyDarker: 'grey-darker',
        },
      },
    };
  });

  describe('render', () => {
    it('should render the location if location prop value is truthy', () => {
      propsToRender.location = 'London';
      const wrapper = shallow(<Component {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a list of days forecast if list is list prop is not empty', () => {
      propsToRender.list = {
        '2017-02-14': [
          {
            day: '2017-02-14',
            hour: '18:00:00',
            dt_txt: '2017-02-14 18:00:00',
          },
          {
            day: '2017-02-14',
            hour: '21:00:00',
            dt_txt: '2017-02-14 21:00:00',
          },
        ],
        '2017-02-15': [
          {
            day: '2017-02-15',
            hour: '18:00:00',
            dt_txt: '2017-02-15 18:00:00',
          },
        ],
      };
      const wrapper = shallow(<Component {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
