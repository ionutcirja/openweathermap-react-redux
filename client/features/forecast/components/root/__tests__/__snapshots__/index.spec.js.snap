// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast component render should render a list of days forecast if list is list prop is not empty 1`] = `
<section>
  <ul>
    <WithTheme(Component)
      day="2017-02-14"
      hoursList={
        Array [
          Object {
            "day": "2017-02-14",
            "dt_txt": "2017-02-14 18:00:00",
            "hour": "18:00:00",
          },
          Object {
            "day": "2017-02-14",
            "dt_txt": "2017-02-14 21:00:00",
            "hour": "21:00:00",
          },
        ]
      }
      key="2017-02-14"
    />
    <WithTheme(Component)
      day="2017-02-15"
      hoursList={
        Array [
          Object {
            "day": "2017-02-15",
            "dt_txt": "2017-02-15 18:00:00",
            "hour": "18:00:00",
          },
        ]
      }
      key="2017-02-15"
    />
  </ul>
</section>
`;

exports[`Forecast component render should render the location if location prop value is truthy 1`] = `
<section>
  <Title
    colour="grey-darker"
  >
    London
  </Title>
</section>
`;
